### ISSUE TITLE
`solid.utils.offset_points()` returns unexpected endpoints

### ISSUE THREAD

**Issue by etjones** [SolidCode/SolidPython#146]
I'd  expect `solid.utils.offset_points()` to return points offset from a list of points and perpendicular to start & end points. In the below image, the top polygon is expected behavior and the bottom polygon is actual behavior. So... gotta check out what's going wrong there. 

<img width="593" alt="image" src="https://user-images.githubusercontent.com/615213/83906105-8c253300-a728-11ea-9709-60ec68fb609c.png">

Here's the code that generated these shapes. Once this bug is corrected, this code should generate two identical polygons:

```python
#! /usr/bin/env python
from pathlib import Path

from solid import *
from solid.utils import offset_points, euclidify
from euclid3 import Point2

SEGMENTS = 48


def trace_polygon():
    points = [Point2(*p) for p in [(0,0), (5,0), (10,5), (15,5)]]

    # A bug in solid.utils.offset_points() makes this shape look bad
    inside_points = offset_points(points, offset=1, internal=True)
    outside_points = list(reversed(offset_points(points, offset=1, internal=False)))
    all_points =  inside_points + outside_points
    poly1 = polygon(all_points)

    # here's what the offsets *should* do
    manual_points = [
        [0, 1], [4.5857864376, 1.0], [9.5857864376, 6.0], [15,6],
        [15,4], [10.4142135624, 4.0], [5.4142135624, -1.0], [0, -1], 
    ]
    poly2 = translate([0,10])(polygon(manual_points))

    a = poly1 + poly2
    return a

if __name__ == '__main__':
    a = trace_polygon()
    out_path = scad_render_to_file(a, file_header='$fn = %s;' % SEGMENTS, include_orig_code=True)
    print(f'Wrote SCAD file to {out_path}')


```

**Comment by etjones** [comment]
Looks like this is an open-vs-closed polygon thing. Will solve by adding an argument to `offset_points()`

**Comment by etjones** [comment]
Resolved in v0.4.9, fa8f4570ab78e33336312fe39b61890458175f03.  Running the code above now results in two identical shapes
